osidb/models.py

```diff
@@ -1538,22 +1541,28 @@ def __init__(self, *args, **kwargs):
         Flaw, on_delete=models.CASCADE, related_name="snippets", blank=True, null=True
     )
 
-    def convert_snippet_to_flaw(self) -> Union[Flaw, None]:
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_snippets", fields=["source", "external_id"]
+            ),
+        ]
+
+    def convert_nvd_snippet_to_flaw(self) -> Union[Flaw, None]:
```

Comment:
I would keep the original function name as we plan to use this function in all collectors.

Reply:
Exactly. There should be nothing collector specific. If you see it differently, we should adjust the method and not the naming.